### PR TITLE
Fix: Make name optional & add metadata to seismicstore

### PR DESCRIPTION
**Pull request by skairunner**

### CODE REVIEW

**Review by MariuszCognite — CHANGES_REQUESTED** [review]
There is already an optional string type (that we're using) - google.protobuf.StringValue

**Comment by skairunner** [comment]
Oh ok, that's cool

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto [hunk 1]
```diff
@@ -37,6 +37,11 @@ message OptionalMap {
     map<string, string> data = 1;
 }
 
+// Used in edit messages to distinguish between empty strings and missing strings
+message OptionalString {
```
Comment (MariuszCognite) [inline cognite/seismic/protos/v1/seismic_service_datatypes.proto:41]: this can go now ;)

Comment (skairunner) [reply cognite/seismic/protos/v1/seismic_service_datatypes.proto:41]: i thought i deleted that.. i guess i didnt

Comment (skairunner) [reply cognite/seismic/protos/v1/seismic_service_datatypes.proto]: i never hit cmd s

**Review by MariuszCognite — APPROVED** [review]

**Pull request merged**